### ISSUE TITLE
Searching by '/' can only be exited by hitting return or esc instead of placing the cursor via mouse

### ISSUE THREAD

**Issue by gallor** [VSCodeVim/Vim#2515]
**Is this a BUG REPORT or FEATURE REQUEST?** (choose one):
Feature request

**What happened**:
I often use the '/' command to jump the editor to a specific method or enum I'm searching for and then click to place the cursor exactly where I want. However when I start typing, it appears to still be in 'search' mode and will add all keystrokes to the '/' search, making the cursor/editor jump away, making me redo my search. Only after hitting return or esc can I exit out of the search. It would be nice to have the search exited (or at least have that as an option flag to enable) whenever a click to place the cursor occurs.

**What did you expect to happen**:
I could search, have the editor jump, then place my cursor directly using my mouse and then start typing. 

**How to reproduce it (as minimally and precisely as possible)**:
Search using '/' and have the cursor jump to match, then place the cursor elsewhere using the mouse, then start typing. The search will be updated with the text instead of allowing you to start typing. 

**Environment**:
- Extension (VsCodeVim) version: 0.11.3
- VSCode version: 1.21.1
- OS version: Mac 10.12.1
